### PR TITLE
Chore 6/services

### LINKED ISSUE #6: Set up services to access data
- [ ] for Supabase
- [ ] for Poster

### PR DESCRIPTION
**Pull request by ANDREYDEN**
## Description

- Set up api routes for poster and supabase communication
- Set up services using [`swr`](https://swr.vercel.app/docs/getting-started) to access poster and supabase
- Add a `poster-employees` page to showcase poster communication
- Add a `NEXT_POSTER_ACCESS_KEY` environment variable

## Issue

Fixes #6

### CODE REVIEW

File: lib/posterService.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    const supabaseEmployees: Partial<definitions['employees']>[] = data.map((e: any, i: number) => ({
+      id: i,
+      first_name: e.name,
+      last_name: e.name
+    }))
+    return supabaseEmployees
+  }
+
+  const employees = toSupabaseSchema(data)
```
Comment (alisondraV) [inline lib/posterService.ts:36]: What's the point of a separate function?

Comment (ANDREYDEN) [reply lib/posterService.ts:36]: cleaner code

Comment (ANDREYDEN) [reply lib/posterService.ts:36]: I guess we can flatten it

File: pages/employees.tsx [hunk 2]
```diff
@@ -4,78 +4,76 @@ import { useRouter } from 'next/router'
...
-    async function deleteEmployee(id: number) {
-        await supabase.from('employees').delete().eq('id', id)
-        getEmployees()
+    async function deleteEmployeeAndReload(id: number) {
+        await deleteEmployee(id)
+        router.reload()
     }
 
-    if (!user || loading)
+    if (!mounted) return (<div></div>)
```
Comment (alisondraV) [inline pages/employees.tsx:59]: Why do we need this?

File: pages/employees.tsx [hunk 3]
```diff
@@ -4,78 +4,76 @@ import { useRouter } from 'next/router'
...
+
+    if (employeesError) {
+        return (
+            <div>Error while fetching employees: {employeesError}</div>
+        )
+    }
 
     return (
-        <div className="flex flex-col items-center justify-center py-2">
+        <div id="asd" className="flex flex-col items-center justify-center py-2">
```
Comment (alisondraV) [inline pages/employees.tsx:76]: What does this id suppose to mean?

File: pages/poster-employees.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import { useUser } from '../lib/hooks'
+import { usePosterGetEmployees } from '../lib/posterService'
+
+const PosterEmployees: NextPage = () => {
+    const router = useRouter()
+    const user = useUser()
+
+    const { employees, employeesLoading, employeesError } = usePosterGetEmployees()
+    
+
```
Comment (alisondraV) [inline pages/poster-employees.tsx:14]: ))

File: pages/poster-employees.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    const handleLogOut = async () => {
+        await supabase.auth.signOut()
+        router.replace('/')
+    }
+    
+    if (employeesLoading) {
+        return (
+            <div id="loader" className="flex justify-center items-center">
+                <div className="animate-spin rounded-full h-32 w-32 border-t-2 border-b-2 border-blue-500 mt-3"></div>
+            </div>
```
Comment (alisondraV) [inline pages/poster-employees.tsx:24]: Extract this into a component?

File: pages/employees.tsx (hunk 2 above)
Comment (ANDREYDEN) [reply pages/employees.tsx:59]: because otherwise the whole page will rotate together with a spinner

Comment (ANDREYDEN) [reply pages/employees.tsx:59]: you can try removing it - it will give you weird errors in the console

Comment (ANDREYDEN) [reply pages/employees.tsx:59]: this was a solution from SO

Comment (ANDREYDEN) [reply pages/employees.tsx:59]: this was a solution from SO

File: pages/employees.tsx (hunk 3 above)
Comment (ANDREYDEN) [reply pages/employees.tsx:76]: yeah, this is from testing, I should remove this

File: pages/poster-employees.tsx (hunk 5 above)
Comment (ANDREYDEN) [reply pages/poster-employees.tsx:24]: not in this PR

**Pull request merged**